### PR TITLE
Pt 159406473 sc continuous chain monitoring

### PR DESCRIPTION
**Pull request by uwiger**
Addresses [PT #159406473](https://www.pivotaltracker.com/story/show/159406473) and [#159406302](https://www.pivotaltracker.com/story/show/159406302). Work in progress. Missing e.g. some test case(s) in `aehttp_integration_SUITE`.

### CODE REVIEW

File: apps/aechannel/test/aesc_fsm_SUITE.erl [hunk 1]
```diff
@@ -1224,6 +1238,21 @@ shutdown_(#{fsm := FsmI, channel_id := ChannelId} = I, R) ->
     verify_close_mutual_tx(SignedTx, ChannelId),
     ok.
 
+settle_(TTL, MinDepth, #{fsm := FsmI, channel_id := ChannelId} = I, R, Debug) ->
+    ok = rpc(dev1, aesc_fsm, settle, [FsmI]),
+    _ = await_signing_request(settle_tx, I),
+    ct:log("settle_tx signed", []),
+    mine_blocks(dev1, TTL + 1, Debug),
```
Comment (lucafavatella) [inline apps/aechannel/test/aesc_fsm_SUITE.erl]: From `channel_detects_close_solo`:
this is mining 11 blocks, and not checking that tx is on chain with at least one confirmation.

I see in [failed run](https://24322-99802036-gh.circle-artifacts.com/1/home/builder/aeternity/_build/test/logs/ct_run.aeternity_ct@localhost.2019-04-22_14.54.08/lib.aechannel.aesc_fsm_SUITE.logs/run.2019-04-22_14.54.08/aesc_fsm_suite.channel_detects_close_solo.html) that this line involves block range 185-195, and that microblock is pushed out by mined key block:
```
*** User 2019-04-22 14:55:44.557 ***
micro block created, Info=#{info =>
                             {mic_block,
                              {mic_header,194,<<>>,
                               <<176,245,36,42,232,165,234,219,46,1,94,228,220,
                                 93,181,253,155,20,142,69,126,251,143,5,29,113,
                                 15,21,17,69,81,91>>,
                               <<176,245,36,42,232,165,234,219,46,1,94,228,220,
                                 93,181,253,155,20,142,69,126,251,143,5,29,113,
                                 15,21,17,69,81,91>>,
...
*** User 2019-04-22 14:55:44.604 ***
key block created, Info=#{info =>
                              {key_block,
                                  {key_header,195,
                                      <<176,245,36,42,232,165,234,219,46,1,94,
                                        228,220,93,181,253,155,20,142,69,126,
                                        251,143,5,29,113,15,21,17,69,81,91>>,
                                      <<176,245,36,42,232,165,234,219,46,1,94,
                                        228,220,93,181,253,155,20,142,69,126,
                                        251,143,5,29,113,15,21,17,69,81,91>>,
...
*** User 2019-04-22 14:55:44.606 ***
aec_conductor:stop_mining() (aeternity_dev1@localhost) -> ok
```

This is confirmed from [node mining log](https://24322-99802036-gh.circle-artifacts.com/1/home/builder/aeternity/_build/test/logs/ct_run.aeternity_ct%40localhost.2019-04-22_14.54.08/lib.aechannel.aesc_fsm_SUITE.logs/run.2019-04-22_14.54.08/log_private/aesc_fsm_SUITE/dev1/log/aeternity_mining.log), where I see that tx gets included in a microblock though then pushed out:
``` 
2019-04-22 14:55:44.554 [info] <0.320.0>@aec_conductor:handle_signed_block:988 Block signed: Height = 194; Hash = 294e3bd0fbffba8eafd77e9813efe4b75e47ee7bae62ba0571d02b154c3ab0d3
2019-04-22 14:55:44.600 [info] <0.320.0>@aec_conductor:handle_mined_block:982 Block mined: Height = 195; Hash = c4ec372f1caa0d7bad286ee47dd14b9ae2b4746952e2091c328302d579d4c72f
```

File: apps/aechannel/test/aesc_fsm_SUITE.erl [hunk 2]
```diff
@@ -1224,6 +1238,21 @@ shutdown_(#{fsm := FsmI, channel_id := ChannelId} = I, R) ->
...
+    SignedTx = await_on_chain_report(I, #{info => channel_closed}, ?TIMEOUT),
```
Comment (lucafavatella) [inline apps/aechannel/test/aesc_fsm_SUITE.erl]: How many confirmations are waited for before reporting this?

And if transaction gets included on chain, and then excluded from chain, does a new report get sent?

And if transaction gets included on chain, then excluded, then included again, are a total of 3 reports sent? (on, off, on)

File: apps/aechannel/test/aesc_fsm_SUITE.erl (hunk 1 above)
Comment (lucafavatella) [reply apps/aechannel/test/aesc_fsm_SUITE.erl]: It seems this is mixing assumption of tx on chain, and lock period elapsed.

Comment (lucafavatella) [reply apps/aechannel/test/aesc_fsm_SUITE.erl]: Stabilization attempt drafted as #2349

**Comment by uwiger** [comment]
Superseded by #2407 - closing

**Pull request closed without merging**